### PR TITLE
deployment: remove unused parameter 'podMap'

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: zhangjie <iamkadisi@163.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by neolit123** [comment]
@kadisi please, change the release note to `NONE`.

**Comment by neolit123** [comment]
please, change the commit title and the title of this PR to:
```
deployment: remove unused parameter 'podMap'
```

**Comment by kadisi** [comment]
@neolit123 please check, thank you

**Comment by neolit123** [comment]
@kadisi 
thanks for the amend.

@enisoc any comments on the removal of `podMap` in this case?
it seems unused in `DeploymentController::getAllReplicaSetsAndSyncRevision()`

/ok-to-test

**Comment by neolit123** [comment]
@kadisi 

seems that some files are not happy about the change.
have a look at the failing `pull-kubernetes-typecheck` and `pull-kubernetes-verify` tests.

run `./hack/verify-all.sh` locally first and resolve the outstanding problems.

thanks.

**Comment by kadisi** [comment]
@neolit123 thank you.

**Comment by neolit123** [comment]
the tests are happy now.
thanks.

/lgtm

**Comment by neolit123** [comment]
@kadisi 
hi, sorry for having to battle with the `pull-kubernetes-e2e-gce-device-plugin-gpu` test which recently had a bug.

it seems that the CI is saying `Pending — Milestone is for a future release and cannot be merged`.
i will leave it to the OWNERS to decide if this PR can be merged now. 
thanks!

cc @janetkuo @mfojtik

**Comment by kadisi** [comment]
@neolit123  thank you!

**Comment by kargakis** [comment]
I'm out this week, maybe @tnozicka can have a look

/cc tnozicka

File: pkg/controller/deployment/sync.go [hunk 1]
```diff
@@ -115,7 +114,7 @@ func (dc *DeploymentController) checkPausedConditions(d *apps.Deployment) error
 //
 // Note that currently the deployment controller is using caches to avoid querying the server for reads.
 // This may lead to stale reads of replica sets, thus incorrect deployment status.
-func (dc *DeploymentController) getAllReplicaSetsAndSyncRevision(d *apps.Deployment, rsList []*apps.ReplicaSet, podMap map[types.UID]*v1.PodList, createIfNotExisted bool) (*apps.ReplicaSet, []*apps.ReplicaSet, error) {
```
Comment (kargakis) [inline pkg/controller/deployment/sync.go]: The comment of this function needs to be updated.

**Comment by kargakis** [comment]
One comment, then lgtm

**Pull request merged**